src/modules/auth/auth.service.ts

```diff
@@ -290,4 +299,45 @@ export class AuthService {
       return newUser;
     }
   }
+
+  async changeEmail(id: string, email: string): Promise<void> {
+    try {
+      const user = await this.usersService.getUserById(id);
```

Comment:
you should have user object from guard where you validate token, you don't need to make query again